ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -315,7 +319,15 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
         Transmission transmission = getTransmission(key);
         try {
           if (key.isConnectable()) {
-            handleConnect(key, transmission);
+            transmission.finishConnect();
+            if (transmission.ready()) {
+              connected.add(transmission.getConnectionId());
+              metrics.selectorConnectionCreated.inc();
+            } else {
+              pendingSslTrans.put(transmission.getConnectionId(),
```

Comment:
We can't generalize it as Unready connections per se. Even those that haven't completed CONNECT unready. What we are interested in are those connections which have reached CONNECT state and yet to complete handshakes.